client/src/components/Protected.tsx

```diff
@@ -1,14 +1,21 @@
 import { useEffect } from "react";
 import { useNavigate } from "react-router-dom";
 import { useLoginContext } from "../contexts/loginContext";
+import { checkStatus } from "@/utils/authService";
 
-const withAuth = (WrappedComponent: React.FC) => {
+const withAuth = (WrappedComponent: React.FC, canRedirect: boolean) => {
```

Comment:
boolean argument is usually an anti-pattern:
https://www.internalfb.com/wiki/Hack_Best_Practices/Avoid_boolean_arguments/

at this point, I recommend reshape the arguments to be an object.
```tsx
const ProtectedEvents = withAuth({'WrappedComponent': Events, 'canRedirect': true})
```